src/game_battler.cpp

```diff
@@ -678,6 +678,11 @@ void Game_Battler::Flash(int r, int g, int b, int power, int frames) {
 	flash.time_left = frames;
 }
 
+const std::vector<lcf::rpg::State*> Game_Battler::GetInflictedStatesOrderedByPriority() const {
+	std::vector<lcf::rpg::State*> state_list = State::GetObjects(GetStates());
+	return State::SortedByPriority(state_list);
+}
```

Comment:
This is more an opinion:
I would not use a new method and simply chain it in the battle class:
``std::vector<lcf::rpg::State*> ordered_states = State::Sorted(State::GetObject(enemy->GetStates()))``

Reply:
Well, I thought that would be a good place to use the new `GetObjects()` methods...